Kitodo/src/test/java/org/kitodo/production/forms/copyprocess/CreateProcessFormIT.java

```diff
@@ -75,65 +81,138 @@ public static void cleanDatabase() throws Exception {
-        underTest.setProcesses(new LinkedList<>(Collections.singletonList(tempProcess)));
-        underTest.getMainProcess().setProject(ServiceManager.getProjectService().getById(1));
-        underTest.getMainProcess().setRuleset(ServiceManager.getRulesetService().getById(1));
-        underTest.getMainProcess().setTitle("title");
+    @AfterEach
+    public void cleanUpAfterEach() throws Exception {
+        if (createdProcess != null && createdProcess.getId() != null) {
+            processService.remove(createdProcess.getId());
+            fileService.delete(URI.create(createdProcess.getId().toString()));
+        }
+        createdProcess = null;
+        setScriptPermissions(false);
+    }
+
+    // Helper to create and initialize a CreateProcessForm with common properties
+    private CreateProcessForm setupCreateProcessForm(String docType) throws Exception {
+        CreateProcessForm form = new CreateProcessForm();
+        form.getProcessDataTab().setDocType(docType);
 
+        Process process = new Process();
+        Workpiece workPiece = new Workpiece();
+        TempProcess tempProcess = new TempProcess(process, workPiece);
+        form.setProcesses(new LinkedList<>(Collections.singletonList(tempProcess)));
+
+        form.getMainProcess().setProject(ServiceManager.getProjectService().getById(1));
+        form.getMainProcess().setRuleset(ServiceManager.getRulesetService().getById(1));
+
+        form.updateRulesetAndDocType(tempProcess.getProcess().getRuleset());
+        return form;
+    }
+
+    // Helper to manage script permissions
+    private void setScriptPermissions(boolean enable) throws Exception {
         File script = new File(ConfigCore.getParameter(ParameterCore.SCRIPT_CREATE_DIR_META));
         if (!SystemUtils.IS_OS_WINDOWS) {
-            ExecutionPermission.setExecutePermission(script);
+            if (enable) {
+                ExecutionPermission.setExecutePermission(script);
+            } else {
+                ExecutionPermission.setNoExecutePermission(script);
+            }
         }
+    }
+
+    @Test
+    public void shouldCreateNewProcess() throws Exception {
+        CreateProcessForm underTest = setupCreateProcessForm("Monograph");
+        underTest.getMainProcess().setTitle("title");
+        setScriptPermissions(true);
         long before = processService.count();
         underTest.createNewProcess();
-        if (!SystemUtils.IS_OS_WINDOWS) {
-            ExecutionPermission.setNoExecutePermission(script);
-        }
+        setScriptPermissions(false);
+
         long after = processService.count();
         assertEquals(before + 1, after, "No process was created!");
 
-        // clean up database, index and file system
-        Integer processId = newProcess.getId();
-        processService.remove(processId);
-        fileService.delete(URI.create(processId.toString()));
+        createdProcess = underTest.getMainProcess();
```

Comment:
The determination of the created process could be moved before the first assertion is executed so you have anytime the right process information before cleaning up. If adjusted here it should be adjusted on the other places too.

Reply:
Thanks, good idea